include/vapor/AnnotationRenderer.h

```diff
@@ -108,9 +109,44 @@ class RENDER_API AnnotationRenderer : public MyBase {
 		double width,
 		std::vector<double> color);
 
-//! Render the domain fram
-	void drawDomainFrame(size_t ts) const;
-
+    void _makeTransformMatrices(
+        const Transform* transform,
+        glm::mat4 &scalingMatrix,
+        glm::mat4 &rotationMatrix,
```

Comment:
I am against using non-const pass by reference in the function argument list as it leads to ambiguous code. What I mean by this is if you are passing a list of parameters, one could be a surprise output. This is a common issue when trying to parse legacy code.

```
func(a, b, c) // b is modified. Surprise!
func(a, &b, c) // I know right away that b will be an output.
```

This is not one of our official coding practices (but I would like to propose it) so feel free to reject this review.